src/targets/cpu/CMakeLists.txt

```diff
@@ -53,46 +54,77 @@ add_library(migraphx_cpu
     target.cpp
     write_literals.cpp
 )
+
+generate_export_header(migraphx_cpu
+    EXPORT_FILE_NAME
+        ${CMAKE_BINARY_DIR}/include/migraphx/cpu/export.h)
```

Comment:
Why do we generate another export header instead of reusing the one from migraphx/export.h?

Reply:
Those are two different APIs. In the context of the target `migraphx_cpu`, the definition of `MIGRAPHX_EXPORT` is `__declspec(dllimport)`. That means we import symbols from the dynamic library `migraphx.dll` to link `migraphx_cpu.dll`. However, the required definition of `MIGRAPHX_CPU_EXPORT` is `__declspec(dllexport)`. That is because we want to export symbols of the dynamic library `migraphx_cpu.dll`. So we cannot use `migraphx/export.h` because it is set to the importing symbols in the context of `migraphx_cpu`.

Reply:
Yea I think it would be better to use `WINDOWS_EXPORT_ALL_SYMBOLS` property on all libraries except the API libraries instead. For out-of-tree backends, we need all symbols available to them not the just the ones we use. All symbols are versioned so there shouldn't be an issue with different versions of migraphx used together.

Reply:
`WINDOWS_EXPORT_ALL_SYMBOLS` has a downside - cannot cause import global data symbols automatically. For those, we still need to use `__declspec(dllimport)`. Anyway, we may end up using the `generate_export_header()` function.